### ISSUE TITLE
Bad performance while using OData client vs using HttpClient on Android

### ISSUE THREAD

**Issue by marcmognol** [dotnet/maui#23185]
### Description

I'm in the process to develop an application for my company. This application will connect to OData endpoints. 
To achieve my goal, I followed the Microsoft documentation https://learn.microsoft.com/en-us/odata/client/getting-started to connect my project to our OData endpoint.

But, I'm experiencing very bad performance while using the application, even in Release mode.

I've got the idea to use classic HttpClient to compare the performance, and HttpClient is 2x faster than OData client.

### Steps to Reproduce

1. Clone the reproduction project repository
2. Run the application with Android emulator
3. Click on HttpClient button (multiple times)
4. Click on ODATA button (multiple times)
5. Compare request response time

See example :
![image](https://github.com/dotnet/maui/assets/8171300/d27cea9a-bf5c-44ab-b4a7-e758e5b2c033)

The behavior is same on physical device.

### Link to public reproduction project repository

https://github.com/marcmognol/Maui.OData.PerformanceIssue/

### Version with bug

8.0.60 SR6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Did you find any workaround?

Yes, by using HttpClient, but proxy generated class with OData generator seems to be easier as we manage thousands of entities.

**Comment by albyrock87** [comment]
Try to use SocketsHttpHandler instead of the default handler.

**Comment by marcmognol** [comment]
Hi @albyrock87 

Thank you for your quick reply and your suggestion but I do have the same result.

Below the way I configured the handler:

```
        builder
            .UseMauiApp<App>()
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
            })
            .Services.AddODataClient("TripPin")
                     .ConfigureODataClient(dsc =>
                     {
                         dsc.BaseUri = new Uri("https://services.odata.org/V4/TripPinServiceRW/");
                     })
                     .AddHttpClient()
                     .ConfigurePrimaryHttpMessageHandler(sp =>
                     {
                         var handler = new SocketsHttpHandler();
                    
                         return handler;
                     })
                     .Services.AddTransient<MyODataRepository>();
```

**Comment by drasticactions** [comment]
I don't believe your issue has to do with the MAUI UI framework. It's either a [dotnet runtime](https://github.com/dotnet/runtime/) issue on Android, or an [OData](https://github.com/OData/odata.net) issue. The MAUI UI framework wouldn't affect the underlying connection like that.

@jonathanpeppers Do you know where this should go?

**Comment by marcmognol** [comment]
I even tried with Simple.OData.Client library and same bad performance.

![image](https://github.com/dotnet/maui/assets/8171300/798e5146-5fa2-47f1-a4a0-494d3e9f30a2)

**Comment by RoiChen001** [comment]
I can repro this issue at Android API 30 on the latest 17.11.0 Preview 2.1 (8.0.60).

**Comment by drasticactions** [comment]
> I even tried with Simple.OData.Client library and same bad performance.
> 

Again, I don't doubt you have performance issues. The problem is that I don't believe your issue is specific to the MAUI UI project. This repo is for the cross-platform MAUI UI framework, and I think your issue is specific to the dotnet runtime or OData itself running on Android, both of which can't be addressed here. That's why verifying if it fails outside of the MAUI UI framework (Ex. Building a .NET Android app `dotnet new android`) is important.

**Comment by jonathanpeppers** [comment]
If using `HttpClient` is fast, but using this library isn't:
* https://github.com/OData/odata.net

^^ Seems like you should file an issue here instead? Doesn't seem like this is a .NET MAUI issue?

You could also profile and see exactly what method is responsible for the slowdown:

* https://aka.ms/profile-maui

**Comment by PureWeen** [comment]
> If using `HttpClient` is fast, but using this library isn't:
> 
> * https://github.com/OData/odata.net
> 
> ^^ Seems like you should file an issue here instead? Doesn't seem like this is a .NET MAUI issue?
> 
> You could also profile and see exactly what method is responsible for the slowdown:
> 
> * https://aka.ms/profile-maui

Closing based on recommendation here
